src/pages/test-report/test-report.effects.ts

```diff
@@ -72,4 +73,38 @@ export class TestReportEffects {
         );
     }),
   );
+
+  @Effect()
+  persistTestReport$ = this.actions$.pipe(
+    ofType(
+      testDataActions.ADD_DRIVING_FAULT,
+      testDataActions.ADD_DANGEROUS_FAULT,
+      testDataActions.ADD_SERIOUS_FAULT,
+      testDataActions.REMOVE_DRIVING_FAULT,
+      testDataActions.REMOVE_SERIOUS_FAULT,
+      testDataActions.REMOVE_DANGEROUS_FAULT,
+      testDataActions.RECORD_MANOEUVRES_SELECTION,
+      testDataActions.ADD_MANOEUVRE_DRIVING_FAULT,
+      testDataActions.ADD_MANOEUVRE_SERIOUS_FAULT,
+      testDataActions.ADD_MANOEUVRE_DANGEROUS_FAULT,
+      testDataActions.REMOVE_MANOEUVRE_FAULT,
+      testDataActions.TOGGLE_CONTROLLED_STOP,
+      testDataActions.CONTROLLED_STOP_ADD_DRIVING_FAULT,
+      testDataActions.CONTROLLED_STOP_ADD_SERIOUS_FAULT,
+      testDataActions.CONTROLLED_STOP_ADD_DANGEROUS_FAULT,
+      testDataActions.CONTROLLED_STOP_REMOVE_FAULT,
+      testDataActions.SHOW_ME_QUESTION_PASSED,
+      testDataActions.SHOW_ME_QUESTION_DRIVING_FAULT,
+      testDataActions.SHOW_ME_QUESTION_SERIOUS_FAULT,
+      testDataActions.SHOW_ME_QUESTION_DANGEROUS_FAULT,
+      testDataActions.SHOW_ME_QUESTION_REMOVE_FAULT,
+      testDataActions.TOGGLE_ECO,
+      testDataActions.TOGGLE_CONTROL_ECO,
+      testDataActions.TOGGLE_PLANNING_ECO,
+      testDataActions.TOGGLE_ETA,
+      testDataActions.TOGGLE_LEGAL_REQUIREMENT,
+    ),
+    delay(1000), // Added a 1 second delay to allow other actions to complete/effects to fire
```

Comment:
What made you choose one second? Seems like the kind of delay I would opt for as well.